aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -314,10 +314,13 @@ void bitwise_xor_kernel(TensorIteratorBase& iter) {
 
 void lshift_kernel(TensorIteratorBase& iter) {
   AT_DISPATCH_INTEGRAL_TYPES(iter.dtype(), "lshift_cpu", [&]() {
-    cpu_kernel(iter,
-      [](scalar_t a, scalar_t b) -> scalar_t {
-        return static_cast<std::make_unsigned_t<scalar_t>>(a) << b;
-    });
+    cpu_kernel_vec(iter,
+        [](scalar_t a, scalar_t b) -> scalar_t {
+          return static_cast<std::make_unsigned_t<scalar_t>>(a) << b;
+        },
+        [](Vectorized<scalar_t> a, Vectorized<scalar_t> b) {
```

Comment:
In these lambdas, we 'pass the `Vectorized<scalar_t>` by value, but then, in all the operations in `vec256_int` and `vec512_int` the arguments are passed by const-ref. Which one's the better one? A priori, I'd say that pass-by-value should be the better one, but I don't know the ABI well enough as to be 100% sure on this.

Tagging a few folks that may know this: @ezyang @peterbell10 @ysiraichi

Reply:
That's a fair point. See this godbolt: https://godbolt.org/z/sEbnnE97x. The assembly seems strictly better when passing by value than when passing by `const&`. I agree that, in the case of oneliners and short lambdas, it doesn't matter, but there's many methods from the `Vectorized` template that are fairly non-trivial, and may or may not be inlined. I reckon we should change them all to take their args by value.

Reply:
By value then